runtime/runtime/src/lib.rs

```diff
@@ -1527,13 +1526,7 @@ impl Runtime {
                 break;
             }
             delayed_receipt_count += 1;
-            let key = TrieKey::DelayedReceipt { index: delayed_receipts_indices.first_index };
-            let receipt: Receipt = get(&state_update, &key)?.ok_or_else(|| {
-                StorageError::StorageInconsistentState(format!(
-                    "Delayed receipt #{} should be in the state",
-                    delayed_receipts_indices.first_index
-                ))
-            })?;
+            let receipt = delayed_receipts.pop(&mut state_update)?.expect("queue is not empty");
```

Comment:
The order of operations is different now in this scope. Previously remove and index update was called after validate, now it's called before. I don't know this well enough to tell if it's ok so just flagging for you :)

Reply:
Yeah that is a good point. It's fine though, it only makes a difference if we exit with `StorageInconsistentState`. In this case, it's unrecoverable and we abort applying the chunk, so all changes in `TrieUpdate` are ignored. (And I think we even panic the entire process at some place)